options.go

```diff
@@ -50,8 +53,12 @@ func RegisterCommandLineOptions(flags *flag.FlagSet) (opts *HmacProxyOpts) {
 	return
 }
 
+// Validate ensures that the HmacProxyOpts configuration is correct and parses
+// some of the values into a useable format. It also sets the Mode member that
+// determines which proxy handler to launch. Collects as many error messages
+// as possible and returns them as a single string via the err return value.
 func (opts *HmacProxyOpts) Validate() (err error) {
-	msgs := make([]string, 0)
+	var msgs []string
```

Comment:
Keep the make slice. Read this: http://stackoverflow.com/questions/25543520/declare-slice-or-make-slice

This is next part is up for discussion:
For all these validate methods, just pass a pointer to the slice. but then you would want to ensure that in each you verify the slice is not null. (it gets hairy then). 
I made a simple [playground link](http://play.golang.org/p/8d8HMRs5CT) to show how I would go about it.
This is only important if memory is of concern. Currently, you're going to pass a copy of the slice to each and return another, possibly, updated copy.